File: test-suite/src/test/java/io/micronaut/docs/function/client/aws/atnamed/AnalyticsClient.java
package io.micronaut.docs.function.client.aws.atnamed;

//tag::clazz[]
import io.micronaut.function.client.FunctionClient;
import jakarta.inject.Named;
@FunctionClient
public interface AnalyticsClient {

    @Named("analytics") // <1>
    String visit(String productId);
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/function/client/aws/atnamed/AnalyticsClient.py
# tag::clazz[]
from abc import ABC, abstractmethod

from jakarta.inject import Named
from micronaut.function.client import FunctionClient


@FunctionClient
class AnalyticsClient(ABC):

    @Named("analytics")  # <1>
    @abstractmethod
    def visit(self, product_id: str) -> str:
        ...
# end::clazz[]
